### DISCUSSION TITLE
R run chunks above doesn't seem to detect errors

### DISCUSSION THREAD

**Discussion by idavydov** [REditorSupport/vscode-R#1426] [Q&A]
Hi,

Not sure if that's a bug or an expected behavior. Let's say you have the following `.Rmd` file:

````
```{r}
a  <- stop()
```

```{r}
x <- a + a
```

```{r}
str(x)
```
````

If you go to the line with `str(x)` and run "R: Run Above Chunks", it will start running all of them, despite the fact that R breaks on the first chunk already.

Is that an expected behavior? Is there an option one could use to stop on the chunk where it breaks?

Basically, I'm trying to understand how to debug large Rmd files, where things could break early in the report.
